File: 프로그래머스/lv1/42862. 체육복/체육복.js
function solution (n, lost, reserve) {
    let answer = n-lost.length;
    let checkLost = lost.filter((el) => !reserve.includes(el))
    let checkReserve = reserve.filter((el) => !lost.includes(el))
    
    answer += lost.length - checkLost.length;
    
    
    checkLost.sort((a,b)=>a-b)
    
    checkLost.forEach((el) => {
        if(checkReserve.length === 0){
            return;
        }
        
        if(checkReserve.includes(el-1)){
            console.log('b',checkReserve)
            checkReserve = checkReserve.filter((l) => l !== el -1 )
            console.log('a',checkReserve)
            answer ++;
        }
        else if (checkReserve.includes(el + 1)){
            console.log('b+1',checkReserve)
                 checkReserve = checkReserve.filter((l) => l !== el+1)
            console.log('a',checkReserve)
                answer++;
        }
    })
    

    return answer;
}

File: 프로그래머스/lv1/42862. 체육복/체육복.py
# 전체 학생 수 2명 30명 
# lost와 reserve에 동일한 번호가 있을 경우 처리햐야한다. 

## 조건 
## 자신보다 1 작거나 클 경우에만 빌려줄 수 있다. 
## 그게 아니면 n -= -1 
## n이 0이면 리턴 



def solution(n, lost, reserve):
    set_reserve = set(reserve)-set(lost)
    set_lost = set(lost) - set(reserve)
    for i in set_reserve :
        if i - 1 in set_lost :
            set_lost.remove(i - 1)
        elif i + 1 in set_lost :
            set_lost.remove(i+1)
    return n - len(set_lost)

        
    return answer